### PR TITLE
TranscodingSample::Launcher: replaced pointers to unique_ptr

### PR DESCRIPTION
**Pull request by mpushki**
for MDP-51933 issue.
Replaced pointers usage with unique_ptr in TranscodingSample::Launcher class

### CODE REVIEW

File: samples/sample_multi_transcode/src/sample_multi_transcode.cpp [hunk 1]
```diff
@@ -1093,7 +1093,7 @@ mfxStatus Launcher::CreateSafetyBuffers()
         {
             pBuffer = new SafetySurfaceBuffer(pPrevBuffer);
```
Comment (mgonchar) [inline samples/sample_multi_transcode/src/sample_multi_transcode.cpp]: you can get rid of tmp variable ```pBuffer```

File: samples/sample_multi_transcode/src/sample_multi_transcode.cpp [hunk 2]
```diff
@@ -448,7 +448,7 @@ mfxStatus Launcher::Init(int argc, msdk_char *argv[])
         pThreadPipeline->startStatus = MFX_WRN_DEVICE_BUSY;
         // set other session's parameters
         pThreadPipeline->implType = m_InputParamsArray[i].libType;
-        m_pThreadContextArray.push_back(pThreadPipeline.release());
+        m_pThreadContextArray.push_back(std::unique_ptr<ThreadTranscodeContext> (pThreadPipeline.release()));
```
Comment (mgonchar) [inline samples/sample_multi_transcode/src/sample_multi_transcode.cpp]: you can do simpler
```
m_pThreadContextArray.push_back(std::move(pThreadPipeline));
```

File: samples/sample_multi_transcode/src/sample_multi_transcode.cpp [hunk 3]
```diff
@@ -310,12 +310,12 @@ mfxStatus Launcher::Init(int argc, msdk_char *argv[])
         sts = pAllocator->Init(m_pAllocParam.get());
         MSDK_CHECK_STATUS(sts, "pAllocator->Init failed");
 
-        m_pAllocArray.push_back(pAllocator.get());
+        m_pAllocArray.push_back(std::move(pAllocator));
         pAllocator.release();
```
Comment (mgonchar) [inline samples/sample_multi_transcode/src/sample_multi_transcode.cpp]: now ```pAllocator``` is in moved from state, no need to ```release```

File: samples/sample_multi_transcode/src/sample_multi_transcode.cpp [hunk 4]
```diff
@@ -1083,17 +1082,16 @@ mfxStatus Launcher::VerifyCrossSessionsOptions()
 mfxStatus Launcher::CreateSafetyBuffers()
 {
     SafetySurfaceBuffer* pBuffer     = NULL;
-    SafetySurfaceBuffer* pPrevBuffer = NULL;
+    // SafetySurfaceBuffer* pPrevBuffer = NULL;
```
Comment (mgonchar) [inline samples/sample_multi_transcode/src/sample_multi_transcode.cpp]: just remove this

Comment (mpushki) [reply samples/sample_multi_transcode/src/sample_multi_transcode.cpp]: Sorry, forgot :)

**Pull request merged**